opencensus-shim/src/main/java/io/opentelemetry/opencensusshim/OpenTelemetrySpanImpl.java

```diff
@@ -0,0 +1,231 @@
+import static io.opentelemetry.opencensusshim.SpanConverter.setDoubleAttribute;
+import static io.opentelemetry.opencensusshim.SpanConverter.setLongAttribute;
+import static io.opentelemetry.opencensusshim.SpanConverter.setStringAttribute;
+
+import com.google.common.base.Preconditions;
+import io.opencensus.trace.Annotation;
+import io.opencensus.trace.AttributeValue;
+import io.opencensus.trace.EndSpanOptions;
+import io.opencensus.trace.Link;
+import io.opencensus.trace.MessageEvent;
+import io.opencensus.trace.Span;
+import io.opencensus.trace.Status;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.api.trace.StatusCode;
+import java.util.EnumSet;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+import javax.annotation.Nonnull;
+
+class OpenTelemetrySpanImpl extends Span implements io.opentelemetry.api.trace.Span {
+  private static final Logger LOGGER = Logger.getLogger(OpenTelemetrySpanImpl.class.getName());
+  private static final EnumSet<Span.Options> RECORD_EVENTS_SPAN_OPTIONS =
+      EnumSet.of(Span.Options.RECORD_EVENTS);
+
+  private final io.opentelemetry.api.trace.Span otelSpan;
+
+  OpenTelemetrySpanImpl(io.opentelemetry.api.trace.Span otelSpan) {
+    super(mapSpanContext(otelSpan.getSpanContext()), RECORD_EVENTS_SPAN_OPTIONS);
+    this.otelSpan = otelSpan;
+  }
+
+  @Override
+  public void putAttribute(String key, AttributeValue value) {
+    Preconditions.checkNotNull(key, "key");
+    Preconditions.checkNotNull(value, "value");
+    value.match(
+        arg -> otelSpan.setAttribute(key, arg),
+        arg -> otelSpan.setAttribute(key, arg),
+        arg -> otelSpan.setAttribute(key, arg),
+        arg -> otelSpan.setAttribute(key, arg),
+        arg -> null);
+  }
+
+  @Override
+  public void putAttributes(Map<String, AttributeValue> attributes) {
+    Preconditions.checkNotNull(attributes, "attributes");
+    for (Map.Entry<String, AttributeValue> attribute : attributes.entrySet()) {
+      putAttribute(attribute.getKey(), attribute.getValue());
+    }
+  }
+
+  @Override
+  public void addAnnotation(String description, Map<String, AttributeValue> attributes) {
+    AttributesBuilder attributesBuilder = Attributes.builder();
+    attributes.forEach(
+        (s, attributeValue) ->
+            attributeValue.match(
+                setStringAttribute(attributesBuilder, s),
+                setBooleanAttribute(attributesBuilder, s),
+                setLongAttribute(attributesBuilder, s),
+                setDoubleAttribute(attributesBuilder, s),
+                arg -> null));
+    otelSpan.addEvent(description, attributesBuilder.build());
+  }
+
+  @Override
+  public void addAnnotation(Annotation annotation) {
+    AttributesBuilder attributesBuilder = Attributes.builder();
+    annotation
```

Comment:
Looks like you can extract a helper for converting oc attributes to otel attributes